File: frontend/src/api/hashtags.ts
import type { HashtagPage } from '@/types';
import { api } from './client';

export async function getHashtagPage(tag: string, page = 1): Promise<HashtagPage> {
  const { data } = await api.get<HashtagPage>(`/hashtags/${encodeURIComponent(tag)}`, {
    params: { page, limit: 24 },
  });
  return data;
}

File: backend/app/routers/hashtags.py
from fastapi import APIRouter, Query

from app.dependencies import DbSession, OptionalUser
from app.schemas.hashtag import HashtagPageOut
from app.services.hashtags import get_hashtag_page
from app.utils.pagination import pagination_params

router = APIRouter(prefix="/hashtags", tags=["hashtags"])


@router.get("/{tag}", response_model=HashtagPageOut)
def hashtag_page(
    tag: str,
    db: DbSession,
    viewer: OptionalUser = None,
    page: int = Query(1, ge=1),
    limit: int = Query(24, ge=1, le=50),
):
    page, limit, _ = pagination_params(page, limit)
    return get_hashtag_page(db, tag, viewer, page, limit)
